frontend/src/modules/entitydetails/components/EntityDetails.js

```diff
@@ -187,6 +188,17 @@ export class EntityDetailsBase extends React.Component<InternalProps, State> {
         );
     }
 
+    navigateToPath = (path: string) => {
+        const { dispatch } = this.props;
+
+        dispatch(
+            unsavedchanges.actions.check(
+                this.props.unsavedchanges,
+                () => { dispatch(push(path)); }
+            )
+        );
```

Comment:
[rant] This pattern is repeated over a lot of places, and it makes me sad. I wished there was a way to refactor that so it's simpler to use (something like a Python decorator) but I don't know how that can be done. [/rant]

Reply:
Yeah. :/ Or a component lifecycle method.